pkg/scheduler/cache/cache.go

```diff
@@ -1027,10 +1027,17 @@ func (sc *SchedulerCache) processCleanupJob() {
 	defer sc.Mutex.Unlock()
 
 	if schedulingapi.JobTerminated(job) {
-		delete(sc.Jobs, job.UID)
-		metrics.DeleteJobMetrics(job.Name, string(job.Queue), job.Namespace)
-		klog.V(3).Infof("Job <%v:%v/%v> was deleted.", job.UID, job.Namespace, job.Name)
-		sc.DeletedJobs.Forget(obj)
+		if oldJob, found := sc.Jobs[job.UID]; found {
+			oldPgVersion := oldJob.PgUID
+			newPgVersion := job.PgUID
+			klog.V(5).Infof("just add pguid:%v, try to delete pguid:%v", oldPgVersion, newPgVersion)
+			if oldPgVersion == newPgVersion {
```

Comment:
In which case podgroup ID will change? In my opinion, the created podgroup will keep exist and UID is same, has nothing to do with pod creating or deleting in same job.

Reply:
The UID of the podgroup stored as JobInfo is the same as the name of the pg, and it remains unchanged. However, when the number of pods becomes 0, it triggers the deletion of the pg. When the number of pods is non-zero, a new pg will be created again. At this point, the UID of the pg itself changes, making it unique. For a detailed analysis, please refer to the issue https://github.com/volcano-sh/volcano/issues/3374. I have provided a detailed description of the problem scenario there.